### PR TITLE
Kubernetes Response Engine, Part 5: Falcosidekick + Argo

### PR DESCRIPTION
**Pull request by Issif**
Signed-off-by: Issif <issif_github@gadz.org>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind user-interface

/kind content

> /kind translation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area blog

> /area documentation

> /area videos

**What this PR does / why we need it**:

This PR adds the part 5 of the series about Response Engine.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This PR must be merged after #428 (the date will be changed before), the PR is there for review and fixes before.

### CODE REVIEW

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+> * [Kubernetes Response Engine, Part 1 : Falcosidekick + Kubeless]({{< ref "/blog/falcosidekick-reponse-engine-part-1-kubeless" >}})
+> * [Kubernetes Response Engine, Part 2 : Falcosidekick + OpenFaas]({{< ref "/blog/falcosidekick-reponse-engine-part-2-openfass" >}})
+> * [Kubernetes Response Engine, Part 3 : Falcosidekick + Knative]({{< ref "/blog/falcosidekick-reponse-engine-part-3-knative" >}})
+
+----
+
+The Open Source ecosystem is so vibrant, we can propose another way to create a Kubernetes Response Engine based on our magic duo, `Falco` + `Falcosidekick`.
+Today, we'll use two components of the CNCF project `Argo`:
+* [`Argo Events`](https://argoproj.github.io/projects/argo-events), will receive events from `Falcosidekick` and push into it event bus. 
+* [`Argo Workflow`](https://argoproj.github.io/projects/argo), will listen the event bus and then trigger the workflow if some criterias are encountered.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:20]: if certain criteria are encountered

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+Like for previous examples with `Kubeless`, `OpenFaas` and `Knative`, we'll address the situation where a shell is spawned in a pod and we want to remediate that by deleting it.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:22]: Like we did for previous examples

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+Basically, we'll set up this:
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:24]: This is how we will set this up

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      # port to run HTTP server on
+      port: "12000"
+      # endpoint to listen to
+      endpoint: /falco
+      # HTTP request method to allow. In this case, only POST requests are accepted
+      method: POST
+EOF
+```
+
+As expected, we do have a new service which will listen events from `Falcosidekick` on port **12000**:
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:112]: As expected, we now have a new service which will..

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+```shell+
+kubetcl get svc -n argo-events
+
+NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+webhook-falco-eventsource-svc   ClusterIP   10.43.117.26    <none>        12000/TCP                    11m
+```
+
+## Creation of the Sensor
+
+In `Argo Events` architecture, `Sensors` are responsible to listen the Event Bus and triggers *something* if the criterias we set match.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:123]: 'Sensors' are responsible for listening to the Event Bus and triggering *somerthing should the criteria we set match

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+kubetcl get svc -n argo-events
+
+NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+webhook-falco-eventsource-svc   ClusterIP   10.43.117.26    <none>        12000/TCP                    11m
+```
+
+## Creation of the Sensor
+
+In `Argo Events` architecture, `Sensors` are responsible to listen the Event Bus and triggers *something* if the criterias we set match. 
+In our case, our `Sensor`:
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:124]: our Sensor will:

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+* listen only events pushed by **webhoo-falco** `Event Source`
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:125]: listen only for

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+                arguments:
+                  parameters:
+                    - name: falco-event
+                      value: {}
+EOF
+```
+
+## Creation of the Workflow Template
+
+One piece is missing in our `Argo` stack, we mentionned a template above, we logically need to create it too, with the service account it needs.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:224]: There is one piece missing....       we "mentioned"

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+          - name: falco-event
+      container:
+        image: devopps/kubernetes-response-engine-based-on-event-driven-workflow@sha256:22ee203a33fe88f0f99968daebdcea0ca52c8a3d6f7af4c823ed78ac15b7c5db
+        env:
+          - name: BODY
+            value: "{{inputs.parameters.falco-event}}"
+EOF
+```
+
+`Argo Workflow` runs all workflow steps inside their own pods, we'll use for this tutorial a *Golang* image developped by [@developer-guy](https://github.com/developer-guy) (who wrote the [Part 2]({{< ref "/blog/falcosidekick-reponse-engine-part-2-openfass" >}}) of this serie 😄).
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md]: we will use for this tutorial a ... image "developed"

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+At this stage, everything is ready to receive events from `Falco` and protect our cluster.
+If you go in `Argo Workflow` UI you will find the architecture we described at beginning.
+
+![Event Flow for our Response Engine](/img/argo-workflow-response-engine-event-flow.png)
+
+## Installation of Falco and Falcosidekick
+
+Last but not least, it's time to install our beloved `Falco` and `Falcosidekick` and connect them to our shiny new Response Engine.
+
+Like for other posts of this serie we'll use `Helm` as conveniant installation method.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:296]: As with other posts of this series

File: content/en/blog/falcosidekick-reponse-engine-part-5-argo.md [hunk 11]
```diff
@@ -0,0 +1,377 @@
...
+
+helm repo add falcosecurity https://falcosecurity.github.io/charts
+
+helm install falco falcosecurity/falco \
+  --namespace falco \
+  --set falcosidekick.enabled=true \
+  --set falcosidekick.config.webhook.address=http://webhook-falco-eventsource-svc.argo-events.svc.cluster.local:12000/falco
+```
+
+You remember the service we mentionned earlier? This is it we use, in its FQDN format, as endpoint.
```
Comment (danpopSD) [inline content/en/blog/falcosidekick-reponse-engine-part-5-argo.md:309]: Remember the service we "mentioned" earlier? This is it in its FQDN format as an endpoint.

**Review by danpopSD — COMMENTED** [review]
typos and grammar but some are snippets of the beginning of sentences etc

**Review by leogr — APPROVED** [review]

**Pull request merged**